perun-core/src/main/java/cz/metacentrum/perun/core/entry/MembersManagerEntry.java

```diff
@@ -1204,6 +1204,31 @@ public RichMember sponsorMember(PerunSession session, Member sponsored, User spo
 		return membersManagerBl.getRichMember(session, membersManagerBl.sponsorMember(session, sponsored, sponsor));
 	}
 
+	@Override
+	public List<RichMember> getSponsoredMembers(PerunSession sess, Vo vo, User user, List<String> attrNames) throws InternalErrorException, AttributeNotExistsException, PrivilegeException, VoNotExistsException, UserNotExistsException {
+		Utils.checkPerunSession(sess);
+		perunBl.getVosManagerBl().checkVoExists(sess, vo);
+		perunBl.getUsersManagerBl().checkUserExists(sess, user);
+
+		if (!(AuthzResolver.isAuthorized(sess, Role.REGISTRAR)||AuthzResolver.isAuthorized(sess, Role.VOADMIN, vo))) {
+			throw new PrivilegeException(sess, "getSponsoredRichMembers");
+		}
+
+		List<AttributeDefinition> attributeDefinitions = new ArrayList<>();
+		for (String attrName : attrNames) {
+			attributeDefinitions.add(getPerunBl().getAttributesManagerBl().getAttributeDefinition(sess, attrName));
+		}
+
+		//Basic rich Members without attributes
+		List<RichMember> richMembers = membersManagerBl.convertMembersToRichMembers(sess, membersManagerBl.getSponsoredMembers(sess, vo, user));
+		//Enriched rich members with attributes by list of attributes
+		richMembers = membersManagerBl.convertMembersToRichMembersWithAttributes(sess, richMembers, attributeDefinitions);
+		//RichMembers with filtered attributes by rights from session
+		richMembers = membersManagerBl.filterOnlyAllowedAttributes(sess, richMembers);
```

Comment:
I would prefer using ``filterOnlyAllowedAttributes(PerunSession sess, List<RichMember> richMembers, Group group, boolean useContext)`` with group = NULL and useContext = TRUE.

Its optimized to check rights for first member and store results based on attribute name. If members are from the same VO, you can apply same rights to their member/user attributes. If they would be from different VO, exception is thrown.
Group is optional and not useful for this use-case.
It saves up "init" part of attribute rights check method, where it checks if attribute definition exists etc. which adds overhead for each member and his attribute returned.

Reply:
Fixed. Thank you.